### PR TITLE
8260338: Some fields in HaltNode is not cloned

### PR DESCRIPTION
**Pull request by YaSuenag**
I got strange log as following. Contents in `CodeString` is garbled. It seems not to be initialized.

```
[3.155s][trace][codestrings ] Created CodeString [ dｴ・] (0x7f3804a150)
```

`HaltNode` has two fields - `_reachable` and `_halt_reason`, but they would not be cloned at Node::clone.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8260338](https://bugs.openjdk.java.net/browse/JDK-8260338): Some fields in HaltNode is not cloned

### Reviewers
 * [Xin Liu](https://openjdk.java.net/census#xliu) (@navyxliu - no project role)
 * [Nils Eliasson](https://openjdk.java.net/census#neliasso) (@neliasso - **Reviewer**)
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2213/head:pull/2213`
`$ git checkout pull/2213`

### CODE REVIEW

File: src/hotspot/share/opto/node.cpp [hunk 1]
```diff
@@ -568,6 +568,10 @@ Node *Node::clone() const {
   if (n->is_SafePoint()) {
     n->as_SafePoint()->clone_replaced_nodes();
   }
+  if (n->is_Halt()) {
```
Comment (navyxliu) [inline src/hotspot/share/opto/node.cpp:571]: This line seems unnecessary. I see HaltNode::size() increases from 0x60 to 0x70 with your patch.  Memcpy at line 502 covers it, doesn't it?

Comment (YaSuenag) [reply src/hotspot/share/opto/node.cpp:571]: `_reachable` and `_halt_reason` are available on `HaltNode`, not `Node`,  so I think this condition is necessary.
L502 uses the result of `Node::size_of()`, and I overrided it in `HaltNode`. So memory allocation at L501 and memcpy at L502 are safety.

Comment (navyxliu) [reply src/hotspot/share/opto/node.cpp:571]: size_of() is a virtual function.  line 500 should use HaltNode::size_of(), which give you 112(0x70) bytes, right?
I mean memcpy at line 502 should cover HaltNode's member variables.  
`Copy::conjoint_words_to_lower((HeapWord*)this, (HeapWord*)n, s);`

Comment (YaSuenag) [reply src/hotspot/share/opto/node.cpp]: Ah, you are right, thanks!
I reverted the change for node.cpp.

**Review by navyxliu — APPROVED** [review]
LGTM, but still need a reviewer's approval.

**Review by neliasso — APPROVED** [review]
Looks good.

**Review by TobiHartmann — APPROVED** [review]

**Comment by TobiHartmann** [comment]
@YaSuenag For future PRs, please wait for at least 24 hours before integrating a non-trivial change. A change is only considered trivial if it's explicitly stated by a reviewer in the PR (see **trivial** in [OpenJDK Developers’ Guide](https://openjdk.java.net/guide/#glossary). Thanks.

**Comment by YaSuenag** [comment]
> @YaSuenag For future PRs, please wait for at least 24 hours before integrating a non-trivial change. A change is only considered trivial if it's explicitly stated by a reviewer in the PR (see **trivial** in [OpenJDK Developers’ Guide](https://openjdk.java.net/guide/#glossary). Thanks.

Sorry, I got it.

**Pull request closed without merging**